src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -262,5 +263,44 @@ private static Uri DetermineBaseAddress(string nationalCloud, string version)
             return new Uri(cloudAddress);
 
         }
+
+        internal static TelemetryHandler GetConfiguredTelemetryHandler(GraphClientOptions graphClientOptions)
+        {
+            // Build the service library string from the options
+            var serviceLibraryString = string.Empty;
+            if (!string.IsNullOrEmpty(graphClientOptions?.GraphServiceLibraryClientVersion))
+            {
+                serviceLibraryString = graphClientOptions?.GraphProductPrefix ?? "graph-dotnet";
```

Comment:
what about moving the header build-up logic to the handler itself?

Reply:
Thanks for pointing this out. This was inconsistent with other SDKs(e.g Go and the spec here), and should be resolved in the latest update.